core/app/models/spree/address/state_normalizer.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Spree
+  class Address::StateNormalizer
+    attr_reader :address
+    delegate :state, :state_name, :country, to: :address
+
+    def initialize(address)
+      @address = address
+    end
+
+    def perform!
```

Comment:
Personal preference, but I'd use `def call` here. Did you have a specific reason to use `perform!` instead?

Reply:
I have followed the convention of other service-like objects in Solidus that are using `perform` as their main action method. I have no preferences though.
Probably it's worth defining a project-wise convention for this kind of objects